### ISSUE TITLE
Antigen theme command not working

### ISSUE THREAD

**Issue by desyncr** [zsh-users/antigen#228]
After antigen bundles are loaded from the command line invoking:

    antigen theme <theme>

Does not update the prompt as expected.

**Comment by desyncr** [comment]
I can't seem to reproduce it with latest from develop.

I'm leaving this open to keep an eye on it.

**Comment by jdkbx** [comment]
This is still an issue for me. My custom theme is not applied with versions greater than 1.0.4.

`
#!/usr/bin/env zsh

# ZSH Theme - Preview: http://dl.dropbox.com/u/4109351/pics/gnzh-zsh-theme.png
# Based on bira theme

# load some modules
autoload -U zsh/terminfo # Used in the colour alias below
setopt promptsubst

autoload -U add-zsh-hook

# make some aliases for the colours: (could use normal escape sequences too)
for color in RED GREEN YELLOW BLUE MAGENTA CYAN WHITE; do
  eval PR_$color='%{$fg_bold[${(L)color}]%}'
done
eval PR_NO_COLOR="%{$terminfo[sgr0]%}"
eval PR_BOLD="%{$terminfo[bold]%}"

# Check the UID
if [[ $UID -ne 0 ]]; then # normal user
  eval PR_USER='${PR_GREEN}%n${PR_NO_COLOR}'
  eval PR_USER_OP='${PR_GREEN}%#${PR_NO_COLOR}'
  local PR_PROMPT='$PR_NO_COLOR➤ $PR_NO_COLOR'
else # root
  eval PR_USER='${PR_RED}%n${PR_NO_COLOR}'
  eval PR_USER_OP='${PR_RED}%#${PR_NO_COLOR}'
  local PR_PROMPT='$PR_RED➤ $PR_NO_COLOR'
fi

# Check if we are on SSH or not
if [[ -n "$SSH_CLIENT"  ||  -n "$SSH2_CLIENT" ]]; then
  eval PR_HOST='${PR_YELLOW}%M${PR_NO_COLOR}' #SSH
else
  eval PR_HOST='${PR_GREEN}%M${PR_NO_COLOR}' # no SSH
fi

local return_code="%(?..%{$PR_RED%}%? ↵%{$PR_NO_COLOR%})"

local user_host='${PR_USER}${PR_CYAN}@${PR_HOST}'
local current_dir='%{$PR_BOLD$PR_BLUE%}%~%{$PR_NO_COLOR%}'

function prompt_color() {
  if [[ $UID -ne 0 ]]; then # normal user
    eval PR_USER='${PR_GREEN}%n${PR_NO_COLOR}'
    eval PR_USER_OP='${PR_GREEN}%#${PR_NO_COLOR}'
    echo "%{$PR_NO_COLOR%}➤ %{$PR_NO_COLOR%}"
  else # root
    eval PR_USER='${PR_RED}%n${PR_NO_COLOR}'
    eval PR_USER_OP='${PR_RED}%#${PR_NO_COLOR}'
    echo "%{$PR_RED%}➤ %{$PR_NO_COLOR%}"
fi
}


function ssh_connection() {
  if [[ -n $SSH_CONNECTION ]]; then
    echo "%{$fg_bold[red]%}(ssh) "
  fi
}

PROMPT_SUCCESS_COLOR=$FG[117]
PROMPT_FAILURE_COLOR=$FG[124]
PROMPT_VCS_INFO_COLOR=$FG[242]
PROMPT_PROMPT=$FG[077]
GIT_DIRTY_COLOR=$FG[133]
GIT_CLEAN_COLOR=$FG[118]
GIT_PROMPT_INFO=$FG[012]

${GIT_PROMPT_INFO}$(git_prompt_info)${GIT_DIRTY_COLOR}$(git_prompt_status)${reset_color}${PROMPT_PROMPT}${reset_color}
PROMPT='╭─$(ssh_connection)%{$PR_USER%}%{$PR_CYAN%}@%{$PR_HOST%} %{$PR_BOLD$PR_BLUE%}%~%{$PR_NO_COLOR%}%{$reset_color%} %{$GIT_PROMPT_INFO%}$(git_prompt_info)%{$GIT_DIRTY_COLOR%}$(git_prompt_status) %{$reset_color%}%{$PROMPT_PROMPT%}%{$reset_color%}
╰─$(prompt_color)%{$reset_color%} '
RPS1="${return_code}"

ZSH_THEME_GIT_PROMPT_PREFIX="%{$PR_YELLOW%}‹"
ZSH_THEME_GIT_PROMPT_SUFFIX="%{$GIT_PROMPT_INFO%}›"
ZSH_THEME_GIT_PROMPT_DIRTY=" %{$GIT_DIRTY_COLOR%}✘"
ZSH_THEME_GIT_PROMPT_CLEAN=" %{$GIT_CLEAN_COLOR%}✔"

ZSH_THEME_GIT_PROMPT_ADDED="%{$FG[082]%}✚%{$reset_color%}"
ZSH_THEME_GIT_PROMPT_MODIFIED="%{$FG[166]%}✹%{$reset_color%}"
ZSH_THEME_GIT_PROMPT_DELETED="%{$FG[160]%}✖%{$reset_color%}"
ZSH_THEME_GIT_PROMPT_RENAMED="%{$FG[220]%}➜%{$reset_color%}"
ZSH_THEME_GIT_PROMPT_UNMERGED="%{$FG[082]%}═%{$reset_color%}"
ZSH_THEME_GIT_PROMPT_UNTRACKED="%{$FG[190]%}✭%{$reset_color%}"
`
It is supposed to look somehow like gnzh. But the prompt variable is not set with newer versions of antigen.

**Comment by desyncr** [comment]
I'm not sure if I already ask you but can you [gist](https://gist.github.com) you zsh and antigen (you `$ANTIGEN/.cache/.zcache-payload` would also be great.

**Comment by jdkbx** [comment]
here is the .zshrc
https://gist.github.com/jdkbx/0f28bfd0d1a1a8ee30ed3c7dda3f8c8e

$ANTIGEN/.cache/.zcache-payload does not exist

no other files are changed from their upstream version

**Comment by desyncr** [comment]
Which antigen version are you using? If no cache exists I'm guessing 1.0.x.

**Comment by jdkbx** [comment]
latest master, which is 1.1.2. Even rebooted to check, but no .zcache-payload appears.

**Comment by desyncr** [comment]
@jdkbx Your theme loads correctly here (running `develop`). It does seem to have some character encoding issues, though - not related to antigen.

**Comment by jdkbx** [comment]
when I switch to develop nothing changes. You can ignore the encoding, it works in my terminal, that is not an issue.
I only came about this problem, when I set up a new computer, so antigen and the bundles were freshly cloned from github. But after updating antigen I see the same problem on gentoo and ubuntu with reverting back to 1.0.4 as a possible workaround. If it works on your system, and my only changes are in .zshrc, I suspect the problem in there.

**Comment by jdkbx** [comment]
I just tried on a new account with autogenerated default .zshrc (zsh 5.1.1 on ubuntu), cloned antigen from github and added the lines to load it. My theme still fails. Others work fine (from oh-my-zsh). No files in .cache, neither under master or develop. So I suspect it to be the theme, but then it should not load on your system ...
So I just got it to load, by copying it into ~/.antigen/repos/~robbyrussels~oh-my-zsh.git/themes/ (the default due to antigen use oh-my-zsh) and loading it manually. All themes I try to load on start up from .zshrc fail. In a running zsh I can load the themes from oh-my-zsh successfully, but not local ones (no matter how I specify the path). So this seems to be an issue with search paths. I suppose some of the caching is still broken.

**Comment by desyncr** [comment]
Do run without caching, or version 1.0.x or version v1. I'm guessing Antigen is not the issue here as I could load it without problems (encoding aside).

**Comment by jdkbx** [comment]
setting _ANTIGEN_CACHE_ENABLED="false" is a workaround. With that everything is back to normal.

Why do you guess that the issue is outside antigen? I have reproducible failure behaviour on quiet different systems (gentoo and ubuntu) that only have antigen in common:
Without _ANTIGEN_CACHE_ENABLED set
1. themes load fine, when they are in the oh-my-zsh folder and are loaded manually in a running zsh
2. multiple themes (even from oh-my-zsh; tested pure and gnzh as well as my own) fail to load on start up from .zshrc (they appear garbled or incomplete)
3. loading a local theme manually fails completely (even themes that load from oh-my-zsh copied to a local directory)

**Comment by desyncr** [comment]
I can not reproduce the behaviour you're describing. Good to know there's a workaround.

**Comment by extink** [comment]
I also have the same issue described by @jdkbx, but the workaround with disabling antigen's cache does not solve the issue. For me this issue happened after commit: bca02d7a0b8f86fa01849f833b837c262763a881. Everything works of the themes, except the prompt will not be change.

### Example (using https://github.com/sindresorhus/pure)
#### Expected (commit 126bb5bf45869b758180c703f1c7687ad14da839 and earlier)
```
~/Programming
❯ pwd
/Users/bobsoppe/Programming
```
#### Actual (commit bca02d7a0b8f86fa01849f833b837c262763a881 and later)
```
~/Programming
bobsoppe@WWSW208-BSO:~/Programming% pwd
/Users/bobsoppe/Programming
```

### Setup
macOS 10.12
zsh 5.2.0
antigen v1.1.3

### Config
```
# Antigen
source ~/.antigen/antigen/antigen.zsh

antigen use oh-my-zsh

antigen bundle aws
antigen bundle brew
antigen bundle common-aliases
antigen bundle docker
antigen bundle git
antigen bundle gradle
antigen bundle osx
antigen bundle ssh-agent
antigen bundle thefuck
antigen bundle tmux

antigen bundle unixorn/autoupdate-antigen.zshplugin
antigen bundle zsh-users/zsh-autosuggestions
antigen bundle zsh-users/zsh-completions
antigen bundle zsh-users/zsh-history-substring-search
antigen bundle zsh-users/zsh-syntax-highlighting

antigen bundle mafredri/zsh-async
antigen bundle sindresorhus/pure

antigen-apply
```

### Test config 1
```
# Antigen
source ~/.antigen/antigen/antigen.zsh

antigen bundle mafredri/zsh-async
antigen bundle sindresorhus/pure

antigen-apply
```

### Test config 2
```
# Antigen
source ~/.antigen/antigen/antigen.zsh

antigen use oh-my-zsh

antigen bundle mafredri/zsh-async
antigen bundle sindresorhus/pure

antigen-apply
```

**Comment by desyncr** [comment]
@extink Thanks for the info. You gave me quite a few hints where the problem could come from. I'll take a look tonight.

**Comment by extink** [comment]
@desyncr: the strangest thing has happened. With `antigen apply` everything is working again as expected, but with `antigen-apply` it does not work anymore. Even with or without cache enabled.

**Comment by jdkbx** [comment]
same here, now files are appearing under .cache

**Comment by desyncr** [comment]
👍  Glad it worked. I'll be working on a fix for next [1.1.4](https://github.com/zsh-users/antigen/milestone/7).
